src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Hover/HoverEndpoint.cs

```diff
@@ -101,9 +101,21 @@ public void ApplyCapabilities(VSInternalServerCapabilities serverCapabilities, V
             var sourceText = await documentContext.GetSourceTextAsync(cancellationToken).ConfigureAwait(false);
             response.Range = originalAttributeRange.AsRange(sourceText);
         }
-        else if (_documentMappingService.TryMapToHostDocumentRange(codeDocument.GetCSharpDocument(), response.Range, out var projectedRange))
+        else if (positionInfo.LanguageKind == RazorLanguageKind.CSharp)
         {
-            response.Range = projectedRange;
+            if (_documentMappingService.TryMapToHostDocumentRange(codeDocument.GetCSharpDocument(), response.Range, out var projectedRange))
```

Comment:
Moved this inside the `LanguageKind` condition, because whilst its unlikely that we'd get here for a hover on a Html construct, it didn't seem to be completely impossible based on the code, and whilst its unlikely that a Html range would map correctly in the generated C# document, it isn't completely impossible :)